infra/presubmit.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Check code for common issues before submitting."""
+
+import argparse
+import os
+import subprocess
+import sys
+import yaml
+
+_SRC_ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+def _is_project_file(actual_path, expected_filename):
+  """Returns True if actual_path's name is |expected_filename| and is a file
+  that exists and is in in projects/."""
+  if os.path.basename(actual_path) != expected_filename:
+    return False
+
+  if os.path.basename(os.path.dirname(
+      os.path.dirname(actual_path))) != 'projects':
```

Comment:
nit: this assumes that the file is in the root directory of a project, will not work for a subdir. Maybe using `os.path.commonpath` is a better choice, e.g."

```python
_PROJECTS_ROOT = os.path.join(_SRC_ROOT, 'projects')

actual_path = os.path.realpath(actual_path)  # unless it's already an absolute path
if _PROJECTS_ROOT != os.path.commonpath([actual_path, _PROJECTS_ROOT]):
  return False
```

Reply:
ah, looking at how this function is used below, makes the assumption more reliable, maybe add an assert that `os.path.dirname(expected_filename) == ''`

Reply:
I added `chdir` to `_SRC_ROOT` in `main`